src/bundle-manifest.ts

```diff
@@ -201,16 +201,24 @@ export function generateSharedDepsMergeStrategy(minEntrypoints?: number):
  * bundle for an application shell.
  */
 export function generateShellMergeStrategy(
-    shell: UrlString, minEntrypoints: number): BundleStrategy {
+    shell: UrlString, minEntrypoints?: number): BundleStrategy {
   return (bundles: Bundle[]): Bundle[] => {
+    if (minEntrypoints === undefined) {
+      minEntrypoints = 2;
+    }
     const newBundles = generateSharedDepsMergeStrategy(minEntrypoints)(bundles);
-    const shellBundle = newBundles.find((bundle) => bundle.files.has(shell));
-    const sharedBundle =
-        newBundles.find((bundle) => bundle.entrypoints.size > 1);
-    if (shellBundle && sharedBundle && shellBundle !== sharedBundle) {
+    let shellBundle = newBundles.find(bundle => bundle.files.has(shell));
+    if (!shellBundle) {
+      throw new Error(`No bundle found containing specified shell ${shell}`);
+    }
+    const sharedBundles = newBundles.filter(
+        (bundle) => bundle.entrypoints.size >= minEntrypoints &&
+            bundle !== shellBundle);
+    for (const sharedBundle of sharedBundles) {
       newBundles.splice(newBundles.indexOf(shellBundle), 1);
       newBundles.splice(newBundles.indexOf(sharedBundle), 1);
-      newBundles.push(mergeBundles([shellBundle, sharedBundle]));
+      shellBundle = mergeBundles([shellBundle, sharedBundle]);
```

Comment:
Ok, may be a false alarm, but the loop implies there may be more than one of these... and merging all the shared bundles into the shell isn't what I'd expect. But remembering the shared deps strategy, there should only be one bundle that satisfies `bundle.entrypoints.size >= minEntrypoints` I thought.

Looking at the code for `generateSharedDepsMergeStrategy` though, I'm finding it hard to reason about the `!getBundleEntrypoint(bundle)` exception for putting things in the shared bundle - that could use a comment. I think the only way that an entrypoint bundle could run have more than one entrypoint is if there were a dependency from one entrypoint to another. Either way, if there are more than 1 bundles with `bundle.entrypoints.size >= minEntrypoints`, it seems only one would also satisfy `!getBundleEntrypoint(bundle)` - should the ones that have an entrypoint be excluded from the shell bundle?